multiprocessing_logging.py

```diff
@@ -59,18 +59,23 @@ def setFormatter(self, fmt):
         self.sub_handler.setFormatter(fmt)
 
     def _receive(self):
-        while not (self._is_closed and self.queue.empty()):
-            try:
-                record = self.queue.get(timeout=0.2)
-                self.sub_handler.emit(record)
-            except (KeyboardInterrupt, SystemExit):
-                raise
-            except EOFError:
-                break
-            except queue.Empty:
-                pass  # This periodically checks if the logger is closed.
-            except:
-                traceback.print_exc(file=sys.stderr)
+        try:
+            while not (self._is_closed and self.queue.empty()):
+                try:
+                    record = self.queue.get(timeout=0.2)
+                    self.sub_handler.emit(record)
+                except (KeyboardInterrupt, SystemExit):
+                    raise
+                except (EOFError, BrokenPipeError):
+                    break
+                except queue.Empty:
+                    pass  # This periodically checks if the logger is closed.
+                except:
+                    traceback.print_exc(file=sys.stderr)
+        except BrokenPipeError:
+            pass
+        except:
+            traceback.print_exc(file=sys.stderr)
```

Comment:
I didn't expect `empty()` to fail. Thanks for the report.

I don't like the duplication of the handlers. If you could refactor the code to avoid that, it would be great.
Otherwise, I'll do it as soon as I can.

Reply:
The duplication of the handlers seemed a good idea to me since BrokenPipeError shouldn't bring a traceback on screen, but any other should.
Mostly the same as in the while loop you have EOFError error and a generic except.

Reply:
That behaviour is good and should be kept but not the duplication.

Reply:
I just tried my code again:
- with a double BrokenPipeError catcher (in the while loop and outside).
- with a simple BrokenPipeError catcher outside of the while loop

As soon as I remove the BrokenPipeError catcher inside of the while loop, I get my errors again, so I think I'm stuck with the duplicate exception catcher.

I've done a test on both BrokenPipeErrors, adding a `print('inside')` statement whenever it's triggered inside the loop, and a `print('outside')` statement for the global exception.
Here's my result
```
inside
outside
inside
inside
inside
inside
inside
inside
inside
inside
outside
inside
inside
inside
inside
outside
```

As seen, I get a lot of BrokenPipeErrors in both cases. I have no idea why the outside exception catcher isn't sufficient.

I've tried (again) to make a simple test case, but this seems hard to catch.
Since I almost recreated all parts of my program at the end, I can only come to the conclusion that it has todo with cpu/io intensive tasks as my program does both.

Anyway, for multiprocessing_logging to work for me under Windows, I need that duplicate.

Reply:
Actually, thinking of it, if I remove the inner BrokenPipeError catcher, the one outside the loop will indeed catch the error. But, doing so, it interrupted the while loop which could perhaps continue.

Reply:
Any news on this ?

Reply:
I'll finish implementing a fix for this tomorrow. Sorry for the delay.

Reply:
No probs ;) Just hoping to contribute